MainProject/Company.cs

```diff
@@ -4,44 +4,66 @@ namespace MainProject
 {
     public class Company
     {
-        public string name;
-        public int foundationYear;
-        public Shop[] shops;
+        private Shop[] shops;
+        public readonly string vatNumber;
 
-        public Company()
+        public string Name { get; private set; }
+
+        public int FoundationYear { get; private set; }
+
+        public Shop[] Shops
+        {
+            get { return shops; }
+            private set { shops = value; }
+        }
+
+        public static int MinimumCarCount { get; private set; }
+
+        public Company(string name, int year, string vatNumber)
         {
+            Name = name;
+            FoundationYear = year;
+            this.vatNumber = vatNumber;
         }
 
-        public Company(string name, int year)
+        public Company(string name, int year, Shop shop, string vatNumber)
         {
-            this.name = name;
-            foundationYear = year;
+            Name = name;
+            FoundationYear = year;
+            Shops = new[] { shop };
+            this.vatNumber = vatNumber;
         }
 
-        public Company(string name, int year, Shop shop)
+        public Company(string name, int year, Shop[] shops, string vatNumber)
         {
-            this.name = name;
-            foundationYear = year;
-            shops = new[] { shop };
+            Name = name;
+            FoundationYear = year;
+            Shops = shops;
+            this.vatNumber = vatNumber;
         }
 
-        public Company(string name, int year, Shop[] shops)
+        static Company()
         {
-            this.name = name;
-            foundationYear = year;
-            this.shops = shops;
+            MinimumCarCount = 2;
```

Comment:
I don't see any point of this static field - it's not used in any method for any conditions. But let's leave it as it is

Reply:
Ok